lib/switchtec.c

```diff
@@ -2449,4 +2449,42 @@ int switchtec_get_all_pin_sts(struct switchtec_dev *dev, uint32_t *values)
 	return 0;
 }
 
+/**
+ * @brief Get the RTC counter
+ * @param[in] dev	  Switchtec device handle
+ * @param[in] rtc_counter Pointer to store the RTC counter value
+ * @param[in] operation	  Operation to do on the RTC
+ * @return 0 on success, or a negative value on failure	
+ */
+int switchtec_rtc_counter(struct switchtec_dev *dev, uint64_t *rtc_counter, 
+			  int operation)
```

Comment:
I'd consider using three functions instead of one with an operations flag as that's a bit awkward. 

If you don't want to make that change I'd heavily suggest not abusing the MPRC constants here and use a specific enum that is declared along side this function's prototype. Then  make operations an enum of that type instead of an integer whose usage is entirely unclear.